design/authorization-handler.md

```diff
@@ -0,0 +1,601 @@
+
+In this proposal, we’ll be merging the [Dynamic Scopes discussion](https://github.com/keycloak/keycloak/discussions/8486) and the [RAR Design Proposal](https://github.com/keycloak/keycloak-community/pull/266) into a single place.
+ 
+So many considerations, benefits and implementation details of both proposals can be considered true here, as well as some of the ideas proposed in the [Dynamic Scopes vs RAR discussion](https://github.com/keycloak/keycloak/discussions/8488).
+
+## Authorization Handler VS Client scope
+
+In this design, we are introducing a new entity called Authorization Handler, which is supposed to replace existing Client Scope. There are two possible views how these changes can be seen:
+
+* Introducing new entity "Authorization Handler", which is replacing "Client scope"
+* Extending existing "Client Scope" entity with more capabilities described in this design (dynamic scopes and RAR requests in addition to just “static scope” requests supported by client scopes in current Keycloak)
+
+The overall output should be the same in both cases. From the implementation and backwards compatibility perspective, it might be probably easier to re-use the existing Client Scope entity under the covers, but we are not 100% sure (this will require us to take a look at the current implementation to decide about the best path).
+
+Regarding migration and backwards compatibility, we will need to make sure that existing Client Scopes would be migrated to the Authorization Handler. So users migrating from Keycloak 15.0.2, who have set up some Client Scope will still have exactly the same functionality in new Keycloak versions as before.
+
+In the next text, it can be assumed that Authorization Handler and Client Scope are the same entity for all purposes. So by Client Scope is meant the "Dynamic" client scope, which will be able to handle dynamic scopes and RAR requests. Not just "Static client scope" as available in Keycloak in version 15.0.2.
+
+**Note**: We need community feedback on what the best path would be, and if we decide to go with the authorization handler, it would also be great to get a better name for it.
+
+## The Client Authorization Handler
+
+A new type of entity will be created in Keycloak that will allow users to configure the behaviour of Static, Dynamic Scopes and RAR in a composable way.
+
+The ClientAuthorizationHandler will be a composite structure consisting of the following pluggable components:
+
+* The ClientAuthorizationContext initiator
+* A Compliance processor
+* A consent screen enhancer / builder
+* Configured Protocol Mappers.
+
+
+### ClientAuthorizationContext initiator
+
+In the text below, when it mentions "Initial Authorization-Endpoint request", it is assumed that it can be any of these "initial request" type f orequests. It won't be mentioned again in the further text for the sake of simplicity.
```

Comment:
Minor: Not sure if remove this paragraph and merge it with the following paragraphs, as it seems to be this part is repeated couple of times. This is only cosmetic change, so it is more friendly for readers :-)

Reply:
Oops, it seems like it was mixed up when I was making the change, correcting it.